components/device-mgt/org.wso2.carbon.device.mgt.core/src/main/java/org/wso2/carbon/device/mgt/core/dao/impl/DeviceDAOImpl.java

```diff
@@ -435,6 +435,37 @@ public EnrolmentInfo getEnrolment(DeviceIdentifier deviceId, String currentOwner
         }
     }
 
+    public int getEnrolmentByStatus(DeviceIdentifier deviceId, Status status,
```

Comment:
An Usecase for this method: Lets say we need to have current 'ACTIVE' enrollment of a given device. This method can be used to get it.